homeassistant/components/surepetcare/binary_sensor.py

```diff
@@ -0,0 +1,212 @@
+
+    async_add_entities(entities, True)
+
+
+class SurePetcareBinarySensor(BinarySensorDevice):
+    """A binary sensor implementation for Sure Petcare Entities."""
+
+    def __init__(self, _id: int, name: str,
+                 icon=None, device_class=None, sure_type=None, hass=None):
+        """Initialize a Sure Petcare binary sensor."""
+        self._id = _id
+        self._hass = hass
+        self._name = name
+        self._unit_of_measurement = "%"
+        self._icon = icon
+        self._device_class = device_class
+
+        self._household_id = hass.data[DATA_SURE_PETCARE][CONF_HOUSEHOLD_ID]
+        self._sure_type = sure_type
+
+        self._state = dict()
+        self._data = hass.data[DATA_SURE_PETCARE][sure_type]
+
+    @property
+    def is_on(self):
+        """Return true if entity is on/unlocked."""
+        return bool(self._state)
+
+    @property
+    def should_poll(self):
+        """Return true."""
+        return False
+
+    @property
+    def name(self):
+        """Return the name of the device if any."""
+        return self._name
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the device."""
+        return self._state
+
+    @property
+    def device_class(self):
+        """Return the device class."""
+        return (DEFAULT_DEVICE_CLASS
+                if not self._device_class
+                else self._device_class)
+
+    @property
+    def icon(self):
+        """Return the device class."""
+        return DEFAULT_ICON if not self._icon else self._icon
+
+    @property
+    def unique_id(self):
+        """Return an unique ID."""
+        return "{}-{}".format(self._household_id, self._id)
+
+    async def async_update(self):
+        """Get the latest data and update the state."""
+        try:
+            self._state = self._hass.data[
+                DATA_SURE_PETCARE][self._sure_type][self._id]
+        except (AttributeError, KeyError, TypeError) as error:
+            _LOGGER.debug("async_update error: %s", error)
+
+    async def async_added_to_hass(self):
+        """Register callbacks."""
+        @callback
+        def update():
+            """Update the state."""
+            self.async_schedule_update_ha_state(True)
+
+        # pylint: disable=attribute-defined-outside-init
+        self._async_unsub_dispatcher_connect = async_dispatcher_connect(
+            self._hass, TOPIC_UPDATE, update)
+
+    async def async_will_remove_from_hass(self):
+        """Disconnect dispatcher listener when removed."""
+        if self._async_unsub_dispatcher_connect:
+            self._async_unsub_dispatcher_connect()
+
+
+class Flap(SurePetcareBinarySensor):
+    """Sure Petcare Flap."""
+
+    def __init__(self, _id: int, name: str, hass=None):
+        """Initialize a Sure Petcare Flap."""
+        super().__init__(
+            _id,
+            f"Flap {name.capitalize()}",
+            icon="mdi:lock",
+            device_class="lock",
```

Comment:
Please import the relevant device classes from the binary sensors base component and use here as needed.